src/rulesets/Base/nondiff.jl

```diff
@@ -0,0 +1,358 @@
+@non_differentiable repr(::Any)
+@non_differentiable repr(::MIME, ::Any)
+@non_differentiable repr(::AbstractString, ::Any)
+@non_differentiable reset(::IO)
+@non_differentiable reverse(::AbstractString)
+@non_differentiable rm(::AbstractString)
+@non_differentiable rsplit(::AbstractString)
+@non_differentiable rsplit(::AbstractString, ::AbstractChar)
+@non_differentiable rstrip(::AbstractString)
+
+@non_differentiable seekend(::Any)
+@non_differentiable seekstart(::Any)
+@non_differentiable setenv(::Cmd)
+@non_differentiable show(::Any)
+@non_differentiable show(::Any, ::Any)
+@non_differentiable showerror(::IO, ::Exception)
+@non_differentiable size(::Any)
+@non_differentiable size(::Any, ::Any)
+@non_differentiable sizeof(::Any)
+@non_differentiable split(::AbstractString)
+@non_differentiable split(::AbstractString, ::AbstractChar)
+@non_differentiable splitdir(::AbstractString)
+@non_differentiable splitdrive(::AbstractString)
+@non_differentiable splitext(::AbstractString)
+@non_differentiable splitpath(::AbstractString)
+@non_differentiable startswith(::AbstractString, ::AbstractString)
+@non_differentiable startswith(::AbstractString, ::Regex)
+@non_differentiable stat(::AbstractString)
+@non_differentiable stat(::Base.Filesystem.File)
+@non_differentiable stat(::IOStream)
+@non_differentiable stat(::RawFD)
+@non_differentiable string(::Any)
+@non_differentiable string(::Any, ::Any)
+@non_differentiable string(::Any, ::Any, ::Any)
+@non_differentiable string(::Any, ::Any, ::Any, ::Any)
+@non_differentiable string(::Any, ::Any, ::Any, ::Any, ::Any)
+@non_differentiable strip(::AbstractString)
+@non_differentiable success(::Array{Base.Process,1})
+@non_differentiable success(::Base.Process)
+@non_differentiable success(::Base.ProcessChain)
+@non_differentiable symlink(::AbstractString, ::AbstractString)
+
+@non_differentiable take!(::Base.GenericIOBuffer)
+@non_differentiable take!(::IOStream)
+@non_differentiable tempname(::AbstractString)
+@non_differentiable textwidth(::AbstractChar)
+@non_differentiable textwidth(::AbstractString)
+@non_differentiable titlecase(::AbstractString)
+@non_differentiable titlecase(::AbstractChar)
+@non_differentiable touch(::AbstractString)
+@non_differentiable typemin(::String)
+
+@non_differentiable unescape_string(::AbstractString)
+@non_differentiable unescape_string(::IO, ::AbstractString)
+@non_differentiable unmark(::IO)
+@non_differentiable unsafe_string(::Cstring)
+@non_differentiable uppercase(::AbstractString)
+@non_differentiable uppercase(::AbstractChar)
+@non_differentiable uppercasefirst(::AbstractString)
+
+@non_differentiable wait(::Base.Process)
+@non_differentiable wait(::Base.ProcessChain)
+@non_differentiable wait(::RawFD)
+@non_differentiable write(::IO, ::Any)
+@non_differentiable write(::IO, ::Any, ::Any,)
+@non_differentiable write(::IO, ::Any, ::Any, ::Any)
+@non_differentiable write(::IO, ::Any, ::Any, ::Any, ::Any)
+
+@non_differentiable Libc.free(::Any)
```

Comment:
Looks like `Libc` has quite a lot of methods that certainly don't appear differentiable that don't appear on this list. Is there a particular reason for the omission, or just because you've not had time to write them out yet?

Reply:
We only grabbed once that had particular types uses in their type signatures.
In particular we grabbed nothing with `Number` in the type-sig

Reply:
Oh interesting. Just looking through the methods there are definitely some good candidates e.g. `calloc`.

Reply:
What do you want to do about this one? I'm happy to leave for future work for the sake of getting this in provided that we document what we _think_ is missing in an issue somewhere.